src/mongocxx/pool.cpp

```diff
@@ -39,20 +39,20 @@ void pool::_release(client* client) {
 
 pool::~pool() = default;
 
-pool::pool(const uri& uri, stdx::optional<options::ssl> ssl_options)
+pool::pool(const uri& uri, const options::client& options)
```

Comment:
Most configuration arguments are already in the URI (see [Connection Pool Options](https://docs.mongodb.com/manual/reference/connection-string/#connection-pool-options)).  We only need separate SSL options because there are things like paths that aren't supported by the URI format. I think the likelihood of similar things for pool options are low, so I'm willing to risk an ABI break if we can't find another way to do it.